print-ospf.c

```diff
@@ -708,6 +719,59 @@ ospf_print_ri_lsa_sid_label_range_tlv(netdissect_options *ndo, const uint8_t *tp
     return 0;
 }
 
+static int
+ospf_print_ri_lsa_sid_local_block_tlv(netdissect_options *ndo, const uint8_t *tptr,
```

Comment:
It turns out, when stepping back one step, that the [SID/Label Range TLV](https://datatracker.ietf.org/doc/html/rfc8665#section-3.2) and the [SR Local Block TLV](https://datatracker.ietf.org/doc/html/rfc8665#section-3.3) have exactly the same definition:
```
    0                   1                   2                   3
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |              Type             |             Length            |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |                    Range Size                 |   Reserved    |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |                        Sub-TLVs (variable)                    |
   +-                                                             -+
   |                                                               |
   +                                                               +
```
and each section says *separately* that only the single SID/Label Sub-TLV in section 2.1 is allowed.

So right now, the implementation of `ospf_print_ri_lsa_sid_local_block_tlv()` is identical to that of `ospf_print_ri_lsa_sid_label_range_tlv()`.

Here's the question: do we unify them to one "print subtlvs of local-block or label-range" function, or do we keep them two identical functions (to start with) with the assumption that it's possible that a sub-TLV will be defined only for local-block, or only for label-range?

If we assume they're identical, then the whole local-block printer could just be an additional "case:" label that uses the label-range printer.  But there's nothing in the spec that requires that they be identical.

Any thoughts? @hannesgredler

Reply:
You are right - makes certainly sense to unify. As you have noted encoding is same on both formats.

Reply:
I've unified the case statements in the latest set of changes, thanks.